gamestonk_terminal/cryptocurrency/defi/graph_model.py

```diff
@@ -0,0 +1,271 @@
+         totalLiquidityETH
+        }
+       }
+       """
+    data = query_graph(UNI_URL, query)
+    if not data:
+        return pd.DataFrame()
+    df = pd.Series(data["uniswapFactory"]).reset_index()
+    df.columns = ["Metric", "Value"]
+    df["Value"] = df["Value"].apply(lambda x: very_long_number_formatter(x))
+    return df
+
+
+def get_uniswap_pool_recently_added(
+    last_days: int = 14,
+    min_volume: int = 100,
+    min_liquidity: int = 0,
+    min_tx: int = 100,
+) -> pd.DataFrame:
+    """Get lastly added trade-able pairs on Uniswap with parameters like:
+        * number of days the pair has been active,
+        * minimum trading volume,
+        * minimum liquidity,
+        * number of transactions.
+
+    [Source: https://thegraph.com/en/]
+
+    Parameters
+    ----------
+    last_days: int
+        How many days back to look for added pairs.
+    min_volume: int
+        Minimum volume
+    min_liquidity: int
+        Minimum liquidity
+    min_tx: int
+        Minimum number of transactions done in given pool.
+
+    Returns
+    -------
+    pd.DataFrame
+        Lastly added pairs on Uniswap DEX.
+    """
+
+    days = int(
+        (datetime.datetime.now() - datetime.timedelta(days=last_days)).timestamp()
+    )
+    query = """
+        {
+          pairs(first: 1000,
+          where: {createdAtTimestamp_gt: "%s", volumeUSD_gt: "%s", reserveUSD_gt: "%s", txCount_gt: "%s" },
+          orderBy: createdAtTimestamp, orderDirection: desc) {
+            token0 {
+              symbol
+              name
+            }
+            token1 {
+              symbol
+              name
+            }
+            reserveUSD
+            volumeUSD
+            createdAtTimestamp
+            totalSupply
+            txCount
+          }
+        }
+    """ % (
+        days,
+        min_volume,
+        min_liquidity,
+        min_tx,
+    )
+    data = query_graph(UNI_URL, query)
+    if not data:
+        return pd.DataFrame()
+
+    df = pd.json_normalize(data["pairs"])
+    df["createdAtTimestamp"] = df["createdAtTimestamp"].apply(
+        lambda x: datetime.datetime.fromtimestamp(int(x))
+    )
+
+    df["pair"] = df["token0.symbol"] + "/" + df["token1.symbol"]
+    df.rename(
+        columns={
+            "createdAtTimestamp": "created",
+            "token0.name": "token0",
+            "token1.name": "token1",
+        },
+        inplace=True,
+    )
+    return df[
+        [
+            "created",
+            "pair",
+            "token0",
+            "token1",
+            "volumeUSD",
+            "txCount",
+            "totalSupply",
+        ]
+    ]
+
+
+def get_uni_pools_by_volume() -> pd.DataFrame:
+    """Get uniswap pools by volume. [Source: https://thegraph.com/en/]
+
+    Returns
+    -------
+    pd.DataFrame
+        Trade-able pairs listed on Uniswap by top volume.
+    """
+
+    query = """
+        {
+          pairs(first: 1000, where: {reserveUSD_gt: "1000", volumeUSD_gt: "10000"},
```

Comment:
Can first argument be a parameter or the pools number will never reach this value and it is just to make sure we fetch them all?

Reply:
So it can be a parameter, but the possible range is from 1 to 1000, so I hardcoded it to fetch all of them, 
other way is to add there pagination like skip, limit, but I don't know if user will be interested in some low volume pools :) What do you think ?

Reply:
If possible range is 1000, I think it's ok to leave as you have :)